### ISSUE TITLE
Bug with 2D indexed array literals with optional values: `internal error: missing builtin 'forall'`

### ISSUE THREAD

**Issue by zayenz** [MiniZinc/libminizinc#625]
I get the following error
```
[Playground:3.1-23](err:/private/var/folders/fv/49yy8rf13z5045yfpqbxv3fh0000gn/T/MiniZinc IDE (bundled)-JlgZFp/untitled_model.mzn?line=3&column=1)
  in variable declaration for 'x'
[Playground:4-1.7-3](err:/private/var/folders/fv/49yy8rf13z5045yfpqbxv3fh0000gn/T/MiniZinc IDE (bundled)-JlgZFp/untitled_model.mzn?line=4&column=1)
  in call 'array2d'
  in call 'array2d'
[stdlib_internal:23-5.24-53](err:/Applications/MiniZincIDE.app/Contents/Resources/share/minizinc/std/stdlib/stdlib_internal.mzn?line=23&column=5)
  in let expression
[stdlib_internal:24.8-53](err:/Applications/MiniZincIDE.app/Contents/Resources/share/minizinc/std/stdlib/stdlib_internal.mzn?line=24&column=8)
  in call 'forall'
MiniZinc: evaluation error: internal error: missing builtin 'forall'
Process finished with non-zero exit code 1.
```

when I use a look-up table that uses a 2D indexed array literal that contains optional values. Tested in MiniZincIDE 2.6.4.

Full code showing the problem
```
enum E = {A, B};

array[E, E] of opt E: x = 
[|     A:  B:
 | A:  A,  <>
 | B:  <>, B
 |];
 
output [show(x[A, A])];
```

Using either a non-indexed array literal or not using optional values in the array makes the problem go away.

**Comment by Dekker1** [comment]
Thank you for reporting the issue. The problem with the "missing builtin" already seems to be resolved on the develop branch. With the same fragment I'm still running into an evaluation error issue:

```Error: evaluation error: undeclared identifier '_absent'```

Looking at the stack trace I think somehow the compiler lost track of the fact that the array was optional. I will investigate further when I have time.

**Comment by cyderize** [comment]
This appears to have now been fixed in version 2.7.0
